src/storage/buffer/CMakeLists.txt

```diff
@@ -1,3 +1,25 @@
+include(CheckFunctionExists)
+include(CheckCSourceRuns)
+
+check_function_exists(sched_getcpu HAVE_SCHED_GETCPU)
+if(HAVE_SCHED_GETCPU)
+  add_definitions(-DDUCKDB_HAVE_SCHED_GETCPU)
+endif()
+
+check_c_source_runs(
```

Comment:
We try to avoid constructs like this for various reasons:

* We often don't compile with CMake, but with other build systems, in which case this does not get run
* This doesn't work well when cross-compiling
* In our experience these things are rather brittle and often cause problems in edge cases

Can we use the same rules for `jemalloc` here instead? e.g.:

```cpp
#if defined(__GNUC__)
#define JEMALLOC_HAVE_SCHED_GETCPU
#endif

```

Reply:
It should be:
```
#if defined(_GNU_SOURCE)
#define JEMALLOC_HAVE_SCHED_GETCPU
#endif
```
This is because macOS's clang also defines the `__GNUC__` macro but does not support `sched_getcpu`.